generate-eslint-resolved-configs.js

```diff
@@ -0,0 +1,115 @@
+import { exec } from 'node:child_process';
+import fs from 'node:fs/promises';
+import path from 'node:path';
+
+async function checkPathExists(p) {
+  try {
+    await fs.access(p);
+
+    return true;
+  } catch {
+    return false;
+  }
+}
+
+function sortObjectKeysAlphabetically(obj) {
+  const entries = Object.entries(obj);
+
+  entries.sort((a, b) => a[0].localeCompare(b[0]));
+
+  return Object.fromEntries(entries);
+}
+
+function sortArrayAlphabetically(array) {
+  const clonedArray = [...array];
+  clonedArray.sort((a, b) => a.localeCompare(b));
+
+  return clonedArray;
+}
+
+const NUMBER_TO_SEVERITY_MAP = { 0: 'off', 1: 'warn', 2: 'error' };
+
+function preprocessConfig(configContent) {
+  const config = JSON.parse(configContent);
+
+  config.languageOptions.globals = sortObjectKeysAlphabetically(
+    config.languageOptions.globals
+  );
+
+  config.plugins = sortArrayAlphabetically(config.plugins);
+
+  const rulesEntries = Object.entries(config.rules);
+
+  rulesEntries.sort((a, b) => a[0].localeCompare(b[0]));
+
+  for (const rule of rulesEntries) {
+    const [ruleName, ruleOptions] = rule;
+
+    if (Array.isArray(ruleOptions)) {
+      const severity = ruleOptions[0];
+
+      ruleOptions[0] = NUMBER_TO_SEVERITY_MAP[severity];
+    } else {
+      // This isn't supposed to happen
+      throw new Error(
+        `Rule Options must always be an array, but this is not the case for: ${ruleName}`
+      );
+    }
+  }
+
+  config.rules = Object.fromEntries(rulesEntries);
+
+  return JSON.stringify(config, null, 2);
+}
+
+async function getResolvedConfig(filepath) {
+  return new Promise((resolve, reject) => {
+    exec(`npx eslint --print-config ${filepath}`, (err, stdout, stderr) => {
+      if (err) {
+        reject(err);
+        return;
+      }
+
+      if (stderr) {
+        reject(stderr);
+        return;
+      }
+
+      resolve(stdout);
+    });
+  });
+}
+
+const ESLINT_RESOLVED_CONFIGS_FOLDER = 'eslint-resolved-configs';
+
+(async () => {
+  const eslintPaths = [
+    'config-file.js',
+    'src/file.js',
+    'config-file.ts',
+    'src/file.ts',
+    'declaration-file.d.ts',
+  ];
+
+  const resolvedConfigs = await Promise.all(
+    eslintPaths.map((lintPath) => getResolvedConfig(lintPath))
+  );
+
+  const folderExists = await checkPathExists(ESLINT_RESOLVED_CONFIGS_FOLDER);
+  if (!folderExists) {
+    await fs.mkdir(ESLINT_RESOLVED_CONFIGS_FOLDER);
```

Comment:
Please use this https://stackoverflow.com/questions/13696148/node-js-create-folder-or-use-existing#:~:text=await%20fs.promises.mkdir(dirpath%2C%20%7B%20recursive%3A%20true%20%7D)

instead of hiding the error in `checkPathExists`.

Reply:
Why do you need this anyway? You are creating the folders one by one below.